checker/tests/ainfer-index/non-annotated/ContainsSubstring.java

```diff
@@ -4,12 +4,14 @@
 
 public class ContainsSubstring {
 
-  @SuppressWarnings("samelen") // TODO: caused by a bug related to viewpoint adaptation. Make WPI
-  // actually viewpoint-adapt the annotations that it infers, then
-  // remove this warning suppression.
   public static void run() {
     String word1 = "\"Hamburg\"";
     String word2 = "burg";
+    // The existence of word3 here forces the inferred @SameLen
+    // annotation to include a local variable that isn't a parameter
+    // to compute(), to test that such local variables are viewpoint-adapted
+    // correctly.
+    String word3 = word1;
     System.out.println(compute(word1, word2));
```

Comment:
You need a test case where the arguments don't have the same names as the method parameters.  `delocalize` is resolving the names base on the declaration of `compute` rather than the callsite.

Reply:
Ah, right you are. It was a nice thought while it lasted :)

I will revert back to the previous version and address your previous review directly. Sorry!